controllers/tls_workflow.go

```diff
@@ -1,9 +1,156 @@
 package controllers
 
 import (
+	"github.com/cert-manager/cert-manager/pkg/apis/certmanager"
+	certmanagerv1 "github.com/cert-manager/cert-manager/pkg/apis/certmanager/v1"
 	"github.com/kuadrant/policy-machinery/controller"
+	"github.com/kuadrant/policy-machinery/machinery"
+	"github.com/samber/lo"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	gwapiv1 "sigs.k8s.io/gateway-api/apis/v1"
+
+	kuadrantv1alpha1 "github.com/kuadrant/kuadrant-operator/api/v1alpha1"
+)
+
+const (
+	TLSPolicyAcceptedKey = "TLSPolicyValid"
+)
+
+var (
+	CertManagerCertificatesResource  = certmanagerv1.SchemeGroupVersion.WithResource("certificates")
+	CertManagerIssuersResource       = certmanagerv1.SchemeGroupVersion.WithResource("issuers")
+	CertMangerClusterIssuersResource = certmanagerv1.SchemeGroupVersion.WithResource("clusterissuers")
+
+	CertManagerCertificateKind   = schema.GroupKind{Group: certmanager.GroupName, Kind: certmanagerv1.CertificateKind}
+	CertManagerIssuerKind        = schema.GroupKind{Group: certmanager.GroupName, Kind: certmanagerv1.IssuerKind}
+	CertManagerClusterIssuerKind = schema.GroupKind{Group: certmanager.GroupName, Kind: certmanagerv1.ClusterIssuerKind}
 )
 
-func NewTLSWorkflow() *controller.Workflow {
-	return &controller.Workflow{}
+func NewTLSWorkflow(client *dynamic.DynamicClient) *controller.Workflow {
+	return &controller.Workflow{
+		Precondition:  NewValidateTLSPoliciesValidatorReconciler().Validate,
+		Postcondition: NewTLSPolicyStatusUpdaterReconciler(client).UpdateStatus,
```

Comment:
Are we going to subscribe to TLS-specific events?

```suggestion
		Precondition:  NewValidateTLSPoliciesValidatorReconciler().Subscription().Reconcile,
		Postcondition: NewTLSPolicyStatusUpdaterReconciler(client).Subscription().Reconcile,
```

Reply:
Yes, missed this. Good catch 👍